h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -958,12 +958,18 @@ public void init(boolean expensive) {
       _valid = rebalance(_valid, false, _result + ".temporary.valid");
     }
 
+
     // Drop all non-numeric columns (e.g., String and UUID).  No current algo
     // can use them, and otherwise all algos will then be forced to remove
     // them.  Text algos (grep, word2vec) take raw text columns - which are
     // numeric (arrays of bytes).
     ignoreBadColumns(separateFeatureVecs(), expensive);
     ignoreInvalidColumns(separateFeatureVecs(), expensive);
+
+    if (_response != null && _response.isString()) {
```

Comment:
I think there might be a better way to accomplish this. Because ignoreBad/ignoreInvalid columns are dropping the columns - you can just check if the response is still in the training frame. If it was removed - it was invalid in some sense and it is up to the algorithm to decide what is invalid (the user is free to override ignoreBadColumns)

Also it would be good to put it in the "isSupervised()" section of the init method.